File: modelsqlite/src/main/java/com/zpf/modelsqlite/SQLiteColumn.java
package com.zpf.modelsqlite;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * 数据库表格‘列’的注解
 * Created by ZPF on 2018/1/29.
 */
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.RUNTIME)
public @interface SQLiteColumn {
   @ColumnLimit int column();
}

File: modelsqlite/src/main/java/com/zpf/modelsqlite/anno/SQLiteColumn.kt
package com.zpf.modelsqlite.anno

import com.zpf.modelsqlite.constant.ColumnEnum

/**
 * 数据库表格‘列’的注解
 * Created by ZPF on 2018/1/29.
 */
@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
annotation class SQLiteColumn(val column: ColumnEnum)